lib/jsonapi/matchers/shared.rb

```diff
@@ -0,0 +1,21 @@
+module Jsonapi
+  module Matchers
+    module Shared
+      def normalize_target(target)
+        if target.is_a?(::ActionController::TestResponse)
+          begin
+            JSON.parse(target.body)
+          rescue => e
+            @failure_message = "Expected response to be json string but was #{target.body.inspect}"
```

Comment:
That message will show you there was an error, but not the specific error. It can be helpful to know that there was an unexpected token as a certain position, for example.